eth/stagedsync/stage_log_index.go

```diff
@@ -405,8 +400,11 @@ func pruneLogIndex(logPrefix string, tx kv.RwTx, tmpDir string, pruneTo uint64,
 	logEvery := time.NewTicker(logInterval)
 	defer logEvery.Stop()
 
-	topics := map[string]struct{}{}
-	addrs := map[string]struct{}{}
+	bufferSize := etl.BufferOptimalSize
+	topics := etl.NewCollector(logPrefix, tmpDir, etl.NewAppendBuffer(bufferSize))
```

Comment:
probably oldestEntrySortableBuffer will fit better in your case? because you don't really collect values. AppendBuffer used to collect values list

Reply:
yeah it does, will replace now.